### PR TITLE
create kuberentes-discovery image

### PR DESCRIPTION
**Pull request by deads2k**
Creates an image for `kubernetes-discovery` since this is the API registration, aggregation, and proxy image.

### CODE REVIEW

**Comment by deads2k** [comment]
@mikedanese @cjcullen Where can I check for images?

**Comment by deads2k** [comment]
@k8s-bot cvm gce e2e test this

File: cluster/images/kubernetes-discovery/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2016 The Kubernetes Authors.
```
Comment (mikedanese) [inline cluster/images/kubernetes-discovery/Dockerfile]: A bit confusing but in the diff I sent you, all files in cluster/images/kube-discovery are actually being deleted. Can you drop this directory?

**Comment by mikedanese** [comment]
Can you also add
```
    "kubernetes-discovery": "//cmd/kubernetes-discovery",
```
below this line https://github.com/kubernetes/kubernetes/blob/master/build/BUILD#L8 for parity with our bazel build?

**Comment by mikedanese** [comment]
Otherwise LGTM

**Comment by deads2k** [comment]
Comments addressed.

File: cluster/images/kubernetes-discovery/Dockerfile (hunk 1 above)
Comment (deads2k) [inline cluster/images/kubernetes-discovery/Dockerfile]: > A bit confusing but in the diff I sent you, all files in cluster/images/kube-discovery are actually being deleted. Can you drop this directory?

Deleted, though `cluster/images/kube-discovery` (the old one) still exists in master.  https://github.com/kubernetes/kubernetes/tree/master/cluster/images/kube-discovery

**Comment by deads2k** [comment]
> below this line https://github.com/kubernetes/kubernetes/blob/master/build/BUILD#L8 for parity with our bazel build?

Added.

**Comment by deads2k** [comment]
@k8s-bot node e2e test this

**Pull request merged**